### PR TITLE
CATTY-291 Convert Brick+UserVariable extension to Swift and split up methods

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
Converted `Brick+UserVariable` extension to Swift.

The `isVarOrListBeingUsed` function is split into follow functions:
- isVariableUsed
- isListUsed

Renamed the test class `BrickUserDataTests` to `BrickExtensionTests` 

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc extension Brick {
+    func isVariableUsed(variable: UserVariable) -> Bool {
+        if self is BrickVariableProtocol {
```
Comment (m-herold) [inline src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift:25]: I think this check is not necessary due to the optional cast below.

File: src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            if let variableBrick = self as? (Brick & BrickVariableProtocol) {
+                if variable == variableBrick.userVariable {
+                    return true
+                }
+            }
+        }
+
+        if self is BrickFormulaProtocol {
```
Comment (m-herold) [inline src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift:33]: I think this check is not necessary due to the optional cast below.

File: src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                        }
+                    }
+                }
+            }
+        }
+        return false
+    }
+
+    func isListUsed(list: UserDataProtocol) -> Bool {
+        if self is BrickListProtocol {
```
Comment (m-herold) [inline src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift:48]: I think this check is not necessary due to the optional cast below.

File: src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                if let list = list as? UserList {
+                    if list == listBrick.userList {
+                        return true
+                    }
+                }
+            }
+        }
+
+        if self is BrickFormulaProtocol {
+            if let formulaBrick = self as? BrickFormulaProtocol {
```
Comment (m-herold) [inline src/Catty/Extension&Delegate&Protocol/Extensions/BrickExtension.swift:65]: I think this check is not necessary due to the optional cast below.

File: src/CattyTests/Bricks/BrickExtensionTests.swift [hunk 5]
```diff
@@ -24,7 +24,7 @@ import XCTest
 
 @testable import Pocket_Code
 
-final class BrickUserDataTests: XCTestCase {
+final class BrickExtensionTests: XCTestCase {
```
Comment (m-herold) [inline src/CattyTests/Bricks/BrickExtensionTests.swift:27]: Please rename the names of the test according to the actual method under test, e.g., `testIsVariableUsed`, `testIsListUsedWithReplaceItemInUserListBrick`, etc.).

File: src/CattyTests/Bricks/BrickExtensionTests.swift [hunk 6]
```diff
@@ -51,7 +51,7 @@ final class BrickUserDataTests: XCTestCase {
         let brick = SetVariableBrick()
         brick.setVariable(userList, forLineNumber: 1, andParameterNumber: 1)
 
-        XCTAssertTrue(brick.isVarOrListBeingUsed(userList))
+        XCTAssertTrue(brick.isVariableUsed(variable: userList))
```
Comment (m-herold) [inline src/CattyTests/Bricks/BrickExtensionTests.swift:54]: This should be `brick.isVariableUsed(variable: userList)`.

File: src/CattyTests/Bricks/BrickExtensionTests.swift [hunk 7]
```diff
@@ -187,14 +187,21 @@ final class BrickUserDataTests: XCTestCase {
...
+        XCTAssertTrue(brick.isListUsed(list: userVariable))
     }
 
     func testUserVariableNoListUseNitherVaribleListOrFormulaBrick() {
         let userVariable = UserVariable(name: "testName", isList: false)
 
         let brick = IfLogicEndBrick()
 
-        XCTAssertFalse(brick.isVarOrListBeingUsed(userVariable))
+        XCTAssertFalse(brick.isListUsed(list: userVariable) || brick.isVariableUsed(variable: userVariable))
```
Comment (m-herold) [inline src/CattyTests/Bricks/BrickExtensionTests.swift:205]: Please use two assert statements to make it more intuitive.

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold I have made the changes. Please review

**Comment by m-herold** [comment]
Please resolve merge conflicts.

**Pull request merged**